stub-idp/src/main/java/uk/gov/ida/stub/idp/services/ServiceListService.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.ida.stub.idp.services;
+
+import org.apache.log4j.Logger;
+import uk.gov.ida.jerseyclient.JsonClient;
+import uk.gov.ida.stub.idp.configuration.SingleIdpConfiguration;
+import uk.gov.ida.stub.idp.domain.Service;
+import uk.gov.ida.stub.idp.exceptions.FeatureNotEnabledException;
+
+import javax.inject.Inject;
+import javax.ws.rs.ProcessingException;
+import java.text.MessageFormat;
+import java.util.Optional;
+
+public class ServiceListService {
+
+    private final Logger LOG = Logger.getLogger(ServiceListService.class);
+
+    private final SingleIdpConfiguration singleIdpConfiguration;
+    private final JsonClient jsonClient;
+
+    @Inject
+    public ServiceListService(SingleIdpConfiguration singleIdpConfiguration, JsonClient jsonClient) {
+        this.singleIdpConfiguration = singleIdpConfiguration;
+        this.jsonClient = jsonClient;
+    }
+
+    public Optional<Service[]> getServices() throws FeatureNotEnabledException {
+
+        if (!singleIdpConfiguration.isEnabled()) throw new FeatureNotEnabledException();
+
+        return readListFromHub();
+    }
+
+    private Optional<Service[]> readListFromHub(){
```

Comment:
This should probably not return an Optional. Its more conventional to return something from the Collections lib and handle the fact that it can be of` .size` 0 in the calling class